### PR TITLE
Theme refinements

### PR DESCRIPTION
**Pull request by alex-ketch**
This PR:
- Migrates package builder to Parcel, simplifying configuration and improving build times
- Updates the following themes for compatibility with new HTML encoding, and makes stylistic refinements
  - Stencila
  - Wilmore
  - Tufte
  - eLife (requires [changes from Components package](https://github.com/stencila/designa/pull/289) before release)

There are some discrepancies between the custom selectors generated, and the ones used in the HTML. I'd like to review these with you @nokome, and make changes to the relevant project—either the theme selector generation script, or the HTML encoding.

Some of the selector differences include: 
- `http://schema.org/Table` vs `http://schema.stenci.la/Table`
- `http://schema.org/ItemList ` vs `http://schema.stenci.la/List`
- There are a handful of other `itemprop` vs `data-itemprop` selectors to fix up

### CODE REVIEW

**Comment by nokome** [comment]
Some of the problems are related to this `TODO`: https://github.com/stencila/stencila/blob/efb37aae96a65e59541cef04cc850cc72992a3ce/rust/codec-html/src/encode.rs#L282

But more generally, I think that now might be a good time to review the HTML attributes we are encoding (see https://schema.org/docs/gs.html for reference):

- [ ] Use `https://schema.org` instead of `http://schema.org`
- [ ] Use `data-prop` instead of `data-itemprop` (A little more compact and emphasizes the fact that these are different e.g. `<ol data-prop="authors">` is the array of authors, `<li itemprop="author">` is the individual author)
- [ ] Use a shorter, more similar to `schema.org`, domain e.g. `https://stencila.org` (I think this would be a little more compact and a little more grok-able / easier to understand the parallels to schema.org) 

What do you think?

**Comment by alex-ketch** [comment]
> Use a shorter, more similar to schema.org, domain e.g. https://stencila.org (I think this would be a little more compact and a little more grok-able / easier to understand the parallels to schema.org)

I'm of the opposite opinion on this @nokome, I think mentioning `schema` somewhere in the URL helps signal that this isn't just any old URL, but rather a schema definition.
I really like the other suggestions though 😇

**Comment by nokome** [comment]
OK, I'll make 2 of those changes in this PR so that you can make any necessary changes to the selectors.

**Comment by nokome** [comment]
I've done the following:

- [x] Use `https://schema.org` instead of `http://schema.org`
- [x] Use `https://schema.stenci.la` instead of `http://schema.stenci.la`
- [x] Use `data-root` instead of `data-itemsope="root"` 
- [x] Use `data-prop` instead of `data-itemprop` (A little more compact and emphasizes the fact that these are different e.g. `<ol data-prop="authors">` is the array of authors, `<li itemprop="author">` is the individual author
- [x] Do necessary translation of `itemtype` and `itemprop` based on `@id` properties in the schema

**Comment by nokome** [comment]
@alex-ketch I've made a bunch of changes (see above) including to the generation of selectors. Hopefully that works but could you check please.

Some time we need to migrate things so that the Thema tests use the HTML generated by `rust/codec-html`, rather than Encoda.

File: rust/codec-html/src/encode/data.rs [hunk 1]
```diff
@@ -323,7 +323,7 @@ impl ToHtml for IntegerValidator {
 impl ToHtml for NumberValidator {
     fn to_html(&self, context: &EncodeContext) -> String {
         // We use `.map(|value| value.to_string()` for properties so they get
-        // rendered as text, not wrapped as a `<span itemtype="http://schema.stenci.la/Number"...`
+        // rendered as text, not wrapped as a `<span itemtype="https://schema.stenci.la/Number"...`
```
Comment (nokome) [inline rust/codec-html/src/encode/data.rs:326]: Actually, this should be `https://schema.org/Number`, as should `Boolean` etc (below). I'll make code changes needed to fix this.

**Pull request merged**